### PR TITLE
Fix/spatial mixins

### PR DESCRIPTION
**Pull request by Marius1311**
This implements a batched mode for `correlate` and `impute`, offering a temporary fix for https://github.com/theislab/moscot/issues/729.

### CODE REVIEW

File: src/moscot/problems/space/_mixins.py [hunk 1]
```diff
@@ -414,10 +415,13 @@ def correlate(  # type: ignore[misc]
             - ``'pearson'`` - `Pearson correlation <https://en.wikipedia.org/wiki/Pearson_correlation_coefficient>`_.
             - ``'spearman'`` - `Spearman's rank correlation
                 <https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient>`_.
-        groupby
-            Optional `obs` field in `AnnData` to compute correlations over categorical groups.
         device
             Device where to transfer the solutions, see :meth:`~moscot.base.output.BaseSolverOutput.to`.
+        groupby
+            Optional `obs` field in `AnnData` to compute correlations over categorical groups.
```
Comment (MUCDK) [inline src/moscot/problems/space/_mixins.py:421]: I know this was there before but can we do ` `:attr:`~anndata.AnnData.obs ` `

Comment (MUCDK) [reply src/moscot/problems/space/_mixins.py:421]: :attr:`~anndata.AnnData.obs

**Review by MUCDK — CHANGES_REQUESTED** [review]
Other than the minor doc thing everything looks good to me, thanks Marius!

and in the test maybe once test for this.

**Review by giovp — APPROVED** [review]

File: src/moscot/problems/space/_mixins.py (hunk 1 above)
Comment (Marius1311) [reply src/moscot/problems/space/_mixins.py]: great, done!

**Comment by Marius1311** [comment]
> Other than the minor doc thing everything looks good to me, thanks Marius!
> 
> and in the test maybe once test for this.

Thanks for your feedback @MUCDK, I added that minor change to the docs, extended tests to cover this case, spotted a corner case and fixed it. Everything should be fine now.
